packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -180,14 +208,99 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     });
   }
 
+  // Get indices of selected elements in options array when we have selected values.
+  const getSelectedIndexes = React.useCallback(() => {
+    const indexArray: number[] = [];
+    if (isMultipleEnabled(valueProp, multiple) && !!valueProp) {
+      valueProp.forEach((value) => {
+        indexArray.push(options.findIndex((option) => option.value === value));
+      });
+    }
+    return indexArray;
+  }, [multiple, options, valueProp]);
+
   // Reducer where all the component-wide state is stored
+  const comboBoxReducer = (
+    state: {
+      isOpen: boolean;
+      selectedIndex: number | number[];
+      focusedIndex: number;
+    },
+    {
+      type,
+      value,
+    }:
+      | { type: 'multiselect'; value: number }
+      | { type: 'multi-override'; value: number[] }
+      | { type: 'open'; value: undefined }
```

Comment:
If you remove `value` from these types, you won't need to pass `undefined` to the dispatch.

After that you won't be able to destructure `{ type, value }` above, so you would need to keep the non-destructured version (call it `action` for example) and then access `type` and `value` from inside it (e.g. `type` will become `action.type`).